### PR TITLE
Build the dashboard to show all components and their certs as per the prototype

### PR DESCRIPTION
**Pull request by AmritSidhu**
https://trello.com/c/On8Y3U4B/497-build-the-main-dashboard-showing-all-the-components-and-their-certs-as-per-prototype

Styled the dashboard to follow the designs of the prototype
Hard coded the expiry tag for now just displaying the time the cert was uploaded

### CODE REVIEW

File: app/views/layouts/main_layout.html.erb [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-      <li>
-        <a class="govuk-link" href="#">Manage certificates</a>
-      </li>
-      <li>
-        <a class="govuk-link" href="#">Team Members</a>
-      </li>
-    </ul>
+    <div class="side-navigation">
+      <ul class="govuk-list">
+        <li class="active">
```
Comment (jakubmiarka) [inline app/views/layouts/main_layout.html.erb]: I wouldn't hard code these in the main layout, they should be dynamic, same with the `--no-visited-state` below
```suggestion
        <li>
```

File: app/views/components/index.html.erb [hunk 2]
```diff
@@ -1,5 +1,7 @@
-<h1 class="govuk-heading-xl">Components</h1>
+<h1 class="govuk-heading-l">Manage certificates</h1>
 
-<%= render "sp_components/all_sp_components"%>
+<%= render "shared/msa_certificates"%>
+<%= render "shared/sp_certificates"%>
 
-<%= render "msa_components/all_msa_components"%>
+<!--<%= render "sp_components/all_sp_components"%>
+<%= render "msa_components/all_msa_components"%>-->
```
Comment (brenetic) [inline app/views/components/index.html.erb]: Keeping or removing?

File: app/views/components/index.html.erb [hunk 3]
```diff
@@ -1,5 +1,7 @@
-<h1 class="govuk-heading-xl">Components</h1>
+<h1 class="govuk-heading-l">Manage certificates</h1>
 
-<%= render "sp_components/all_sp_components"%>
+<%= render "shared/msa_certificates"%>
+<%= render "shared/sp_certificates"%>
 
-<%= render "msa_components/all_msa_components"%>
+<!--<%= render "sp_components/all_sp_components"%>
```
Comment (jakubmiarka) [inline app/views/components/index.html.erb]: delete the commented lines?

File: app/assets/stylesheets/components.scss [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Place all the styles related to the components controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
+
+.app-certificate-tag-expiring .govuk-tag {
+  background-color: #b10e1e;
```
Comment (jakubmiarka) [inline app/assets/stylesheets/components.scss]:
```suggestion
  background-color: $govuk-error-colour;
```

Comment (jakubmiarka) [reply app/assets/stylesheets/components.scss]: Something for Chris T, whether red (interpreted as error) is suitable for expiring, whether it shouldn't be yellow as a warning?

File: app/views/components/index.html.erb (hunk 2 above)
Comment (AmritSidhu) [reply app/views/components/index.html.erb]: removed

File: app/views/layouts/main_layout.html.erb (hunk 1 above)
Comment (AmritSidhu) [reply app/views/layouts/main_layout.html.erb]: Do you mean something like this `<li class="<%= 'active' if current_page?(dashboard) %>">`?

Comment (jakubmiarka) [reply app/views/layouts/main_layout.html.erb]: yep!

Comment (AmritSidhu) [reply app/views/layouts/main_layout.html.erb]: With the `--non-visited-state`, we actually want that hard coded in there for the styling to match the prototype. Once the page is visited the link changes colour, but in the prototype they seem to want to keep it the same colour and just bold the current page. This is something i will run by Chris

Comment (jakubmiarka) [reply app/views/layouts/main_layout.html.erb]: Ah yeah, you're right. The guidance says to use the class when `where it is not helpful to distinguish between visited and unvisited states, for example when linking to pages with frequently-changing content such as the dashboard for an admin interface.`

File: app/views/shared/_component_certificates.html.erb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<% if components.present? %>
+  <% components.reverse.each do |component| %>
+  <table class="govuk-table">
+    <caption class="govuk-table__caption">
+      <%= component.component_type == "MsaComponent" ? "Matching Service Adapter" : "Verify Service Provider" %>
```
Comment (jakubmiarka) [inline app/views/shared/_component_certificates.html.erb]:
```suggestion
      <%= component.component_type == COMPONENT_TYPE.MSA ? "Matching Service Adapter" : "Verify Service Provider" %>
```

**Comment by AmritSidhu** [comment]
Chris's comments: change expiring tag to #CB4B0B. add  `govuk-!-margin-bottom-1` class to the table captions. the avoid the thead not being used for anything, add a visually-hidden class to the header, then add some custom styling to put a border on top of the first row

**Comment by christopherthomasdesign** [comment]
To add to that, the `th` elements in the `thead` section should say 'Certificate' and 'Status' on them to refer to the corresponding table columns.

File: app/assets/stylesheets/components.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// Place all the styles related to the components controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
+
+.app-certificate-tag-expiring .govuk-tag {
+  background-color: #CB4B0B;
```
Comment (jakubmiarka) [inline app/assets/stylesheets/components.scss]: that colour is not on the Design System palette: https://design-system.service.gov.uk/styles/colour/ @christopherthomasdesign

File: app/assets/stylesheets/components.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  float: right;
+}
+.side-navigation ul .active {
+  font-weight: bold;
+}
+.side-navigation ul li a {
+  text-decoration: none;
+}
+.app-table__body--noheading { 
+  border-top: 1px solid #bfc1c3
```
Comment (jakubmiarka) [inline app/assets/stylesheets/components.scss]: we shouldn't be using hex codes, but the variables our CSS framework provides
```suggestion
  border-top: 1px solid $govuk-border-colour 
```

File: app/assets/stylesheets/components.scss (hunk 6 above)
Comment (christopherthomasdesign) [reply app/assets/stylesheets/components.scss]: The orange on the palette in the design system doesn't have enough contrast with white text so I darkened it to be WCAG AA compliant. Would be just as happy using the `govuk-error-colour` red but see your point about it maybe being too alarming... Not sure black text on a yellow background would be better in that respect either. I think we have a good enough reason to add a new colour here.

**Comment by jakubmiarka** [comment]
[Build the main "dashboard" showing all the components and their certs (as per prototype)](https://trello.com/c/On8Y3U4B/497-build-the-main-dashboard-showing-all-the-components-and-their-certs-as-per-prototype)

**Pull request merged**